pkg/virt-handler/isolation/isolation.go

```diff
@@ -0,0 +1,206 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package isolation
+
+//go:generate mockgen -source $GOFILE -package=$GOPACKAGE -destination=generated_mock_$GOFILE
+
+/*
+ ATTENTION: Rerun code generators when interface signatures are modified.
+*/
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"syscall"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/log"
+	"kubevirt.io/kubevirt/pkg/virt-handler/cmd-client"
+)
+
+// PodIsolationDetector helps detecting cgroups, namespaces and PIDs of Pods from outside of them.
+// Different strategies may be applied to do that.
+type PodIsolationDetector interface {
+	// Detect takes a vm, looks up a socket based the VM and detects pid, cgroups and namespaces of the owner of that socket.
+	// It returns an IsolationResult containing all isolation information
+	Detect(vm *v1.VirtualMachineInstance) (*IsolationResult, error)
+
+	// Whitelist allows specifying cgroup controller which should be considered to detect the cgroup slice
+	// It returns a PodIsolationDetector to allow configuring the PodIsolationDetector via the builder pattern.
+	Whitelist(controller []string) PodIsolationDetector
+}
+
+type socketBasedIsolationDetector struct {
+	socketDir  string
+	controller []string
+}
+
+// NewSocketBasedIsolationDetector takes socketDir and creates a socket based IsolationDetector
+// It returns a PodIsolationDetector which detects pid, cgroups and namespaces of the socket owner.
+func NewSocketBasedIsolationDetector(socketDir string) PodIsolationDetector {
+	return &socketBasedIsolationDetector{socketDir: socketDir, controller: []string{"cpu", "cpuacct", "cpuset", "freezer", "memory", "net_cls", "perf_event"}}
```

Comment:
I do not sure about controllers, I checked cgroups for one of my containers and I can see that
```
...
5:net_cls,net_prio:/system.slice/docker-382cbb1ae8bcee2ae9308183a9823d54d556cdf143d0b872a32b2c4fd9216e16.scope
...
2:cpu,cpuacct:/system.slice/docker-382cbb1ae8bcee2ae9308183a9823d54d556cdf143d0b872a32b2c4fd9216e16.scope
...
```
and it does not have separate cgroup for `cpu`, `cpuacct` and `net_cls`.

Reply:
Addressed